xsnap/sources/xsnap-worker.c

```diff
@@ -931,6 +943,10 @@ static void xs_issueCommand(xsMachine *the)
 	size_t len;
 	int readError = fxReadNetString(fromParent, &buf, &len);
 	if (readError != 0) {
+		if (feof(fromParent)) {
+			fprintf(stderr, "Got EOF on netstring read. Has parent died?\n");
+			c_exit(E_IO_ERROR);
```

Comment:
I'm wondering if this shouldn't be an `fxAbort` here since we're in the middle of executing JS code?

Not sure what the code should be in that case.

Reply:
```c
fxAbort(the, REASON);
```

...where REASON here is probably `xsFatalCheckExit`. The implementation of `fxAbort` depends on the host. In xsnap, I think that will basically do a printf and then unwind the JavaScript stack by calling `fxExitToHost`, returning control to the host to do a clean exit. You'll want to verify that it does what you want.

Reply:
> returning control to the host to do a clean exit. You'll want to verify that it does what you want.

Yeah that's the part I'm unclear about, on where exactly that goes back to in `xsnap-worker`. Obviously we don't want anything that would cause more pipe I/O at that point, e.g. writing a failing result.

Reply:
I am fine with any error handling, as long as we get a nice short clear message indicating that the error/exit is due to an EOF/broken pipe.

Reply:
Unless we want cleanup to take place within JS, this seems fine to me... but we should document both of these new exits as a breadcrumb for any future changes to accommodate them.